### PR TITLE
A pretty simple telemetry solution for darc.exe

### PR DESCRIPTION
**Pull request by ChadNedzlek**
It reports:
* any command line parsing error as a LogError with relevant information
* any exception that bubbles out to Main
* every verb executed including any options passed and their values
* sensitive options can be redacted with `[RedactFromLogging]`, which record they were specified, but not the values passed

### CODE REVIEW

**Comment by ChadNedzlek** [comment]
@missymessa This might be interesting to add to the little tool I saw you writing

**Comment by ChadNedzlek** [comment]
@mmitche @riarenas Let me know if there are any other options that, when parsed, might be dangerous/useless, so we can slap [RedactFromLogging] on them. :-)

File: src/Microsoft.DotNet.Darc/src/Darc/Program.cs [hunk 1]
```diff
@@ -86,5 +169,105 @@ private static Type[] GetOptions()
...
                 };
         }
+
+        private static void InitializeTelemetry()
+        {
+            var isDebugging = Debugger.IsAttached;
+            var config = TelemetryConfiguration.CreateDefault();
+            if (!isDebugging)
+            {
+                config.InstrumentationKey = "9fcef3c7-f401-41c7-9e91-1f6029c8dcc3";
```
Comment (missymessa) [inline src/Microsoft.DotNet.Darc/src/Darc/Program.cs:179]: Should we stick this GUID in a config file and pull it in with DI options?

Comment (ChadNedzlek) [reply src/Microsoft.DotNet.Darc/src/Darc/Program.cs:179]: We could, but that means we need to make sure the config file makes it through all the layers of moving the exe around... and also hook up the entire config path just for one value. :-)

If we already _had_ a config story, I'd say we'd put this there, but hooking all that up seems overkill for this one value. :-)

Comment (ChadNedzlek) [reply src/Microsoft.DotNet.Darc/src/Darc/Program.cs:179]: For comparison, here is the usage that "dotnet.exe" has. :-) https://github.com/dotnet/sdk/blob/72d3792bb1e5ee56841408009a8638d51acd948b/src/Cli/dotnet/Telemetry/Telemetry.cs#L24

**Pull request merged**